src/control/server/config/server.go

```diff
@@ -49,6 +49,7 @@ type Server struct {
 	DisableVMD          bool                   `yaml:"disable_vmd"`
 	EnableHotplug       bool                   `yaml:"enable_hotplug"`
 	NrHugepages         int                    `yaml:"nr_hugepages"` // total for all engines
```

Comment:
Should this be changed to a uint? A little concerned that existing configs may have the -1 value and I'm not sure what happens in that case.

Reply:
I started changing this to uint but the cascading updates made me think it wasn't a worthwhile change, you end up having to cast things like uint(len(<slice>)) and that doesn't feel nice. We validate that the value is not <0 early in the flow and tbh overall feels better without changing to uint. happy to change back if you feel like it though.